test/module/shared_model/validators/field_validator_test.cpp

```diff
@@ -0,0 +1,747 @@
+          return reason;
+        },
+        role_id_test_cases}},
+      {"detail",
+       {[&] {
+          validation::ReasonsGroupType reason;
+          field_validator.validateAccountDetailKey(reason, detail_key);
+          return reason;
+        },
+        detail_test_cases}},
+      {"tx_hashes",
+       {[&] {
+          validation::ReasonsGroupType reason;
+          field_validator.validateAccountDetailKey(reason, detail_key);
+          return reason;
+        },
+        tx_hashes_test_cases}},
+  };
+};
+
+/**
+ * @given field values from test cases
+ * @when field validator is invoked on command's fields
+ * @then field validator correctly rejects invalid values, and provides
+ * meaningful message
+ */
+TEST_F(FieldValidatorTest, CommandFieldsValidation) {
+  iroha::protocol::Transaction proto_tx;
+  auto payload = proto_tx.mutable_payload();
+
+  // iterate over all commands in transaction
+  iterateContainer(
+      [] { return iroha::protocol::Command::descriptor(); },
+      [&](auto field) {
+        // Add new command to transaction
+        auto command = payload->add_commands();
+        //  // Set concrete type for new command
+        return command->GetReflection()->MutableMessage(command, field);
+      },
+      [this](auto field, auto command) { this->runTestCases(field); },
+      [] {});
+}
+
+/**
+ * @given field values from test cases
+ * @when field validator is invoked on transaction's fields
+ * @then field validator correctly rejects invalid values, and provides
+ * meaningful message
+ */
+TEST_F(FieldValidatorTest, TransactionFieldsValidation) {
+  // iterate over all fields in transaction
+  iterateContainer(
+      [] { return iroha::protocol::Transaction::descriptor(); },
+      [&](auto field) {
+        // generate message from field of transaction
+        google::protobuf::DynamicMessageFactory message_factory;
+        auto field_desc = field->message_type();
+        // will be null if field is not of message type
+        EXPECT_NE(nullptr, field_desc);
+        return message_factory.GetPrototype(field_desc)->New();
+      },
+      [this](auto field, auto transaction_field) { this->runTestCases(field); },
+      [] {});
+}
+
+/**
+ * @given field values from test cases
+ * @when field validator is invoked on query's fields
+ * @then field validator correctly rejects invalid values, and provides
+ * meaningful message
+ */
+TEST_F(FieldValidatorTest, QueryFieldsValidation) {
+  iroha::protocol::Query qry;
+  auto payload = qry.mutable_payload();
+  // iterate over all field in query
+  iterateContainer(
+      [] {
+        return iroha::protocol::Query::Payload::descriptor()->FindOneofByName(
+            "query");
+      },
+      [&](auto field) {
+        // Set concrete type for new query
+        return payload->GetReflection()->MutableMessage(payload, field);
+      },
+      [this](auto field, auto query) { this->runTestCases(field); },
+      [&] {});
+}
```

Comment:
It looks like test for query fields is missing. The purpose is the same as `TransactionFieldsValidation`, since `QueryFieldsValidation` only does validation of concrete query (such as GetAccount) fields. It could iterate `iroha::protocol::Query::descriptor()`.

Reply:
What do you mean by missing? if a field of a query is not covered by tests, an exception will be thrown inside runTestCases()